x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -0,0 +1,426 @@
+  },
+  {
+    name: 'With Eventing',
+    id: 'C2A9093E-E289-4C0A-AA44-8C32A414FA7A',
+  },
+];
+
+const FILE_OPERATIONS: string[] = ['creation', 'open', 'rename', 'execution', 'deletion'];
+
+// These are from the v1 schemas and aren't all valid ECS event categories, still in flux
+const OTHER_EVENT_CATEGORIES: string[] = ['driver', 'file', 'library', 'network', 'registry'];
+
+function randomN(n: number): number {
+  return Math.floor(Math.random() * n);
+}
+
+function* randomNGenerator(max: number, count: number) {
+  while (count > 0) {
+    yield randomN(max);
+    count--;
+  }
+}
+
+function randomArray<T>(lenghtLimit: number, generator: () => T): T[] {
+  const rand = randomN(lenghtLimit) + 1;
+  return [...Array(rand).keys()].map(generator);
+}
+
+function randomMac(): string {
+  return [...randomNGenerator(255, 6)].map(x => x.toString(16)).join('-');
+}
+
+function randomIP(): string {
+  return [10, ...randomNGenerator(255, 3)].map(x => x.toString()).join('.');
+}
+
+function randomVersion(): string {
+  return [6, ...randomNGenerator(10, 2)].map(x => x.toString()).join('.');
+}
+
+function randomChoice<T>(arg: T[]): T {
+  return arg[randomN(arg.length)];
+}
+
+function randomString(length: number): string {
+  return [...randomNGenerator(36, length)].map(x => x.toString(36)).join('');
+}
+
+function randomHostname(): string {
+  return `Host-${randomString(10)}`;
+}
+
+export class EndpointDocGenerator {
+  agentId: string;
+  hostId: string;
+  hostname: string;
+  lastDHCPLeaseAt: number;
+  macAddress: string[];
+  ip: string[];
+  agentVersion: string;
+  os: OSFields;
+  policy: { name: string; id: string };
+
+  constructor() {
+    this.hostId = uuid.v4();
+    this.agentId = uuid.v4();
+    this.hostname = randomHostname();
+    this.lastDHCPLeaseAt = new Date().getTime();
+    this.ip = randomArray(3, () => randomIP());
+    this.macAddress = randomArray(3, () => randomMac());
+    this.agentVersion = randomVersion();
+    this.os = randomChoice(OS);
+    this.policy = randomChoice(POLICIES);
+  }
+
+  public generateEndpointMetadata(ts: number): EndpointMetadata {
+    // If we generate metadata with a timestamp at least 12 hours after the last generated doc,
+    // change the IPs for the host
+    if (Math.abs(ts - this.lastDHCPLeaseAt) > 3600 * 12 * 1000) {
+      this.lastDHCPLeaseAt = ts;
+      this.ip = randomArray(3, () => randomIP());
+    }
+    return {
+      '@timestamp': ts,
+      event: {
+        created: ts,
+      },
+      endpoint: {
+        policy: {
+          id: this.policy.id,
+        },
+      },
+      agent: {
+        version: this.agentVersion,
+        id: this.agentId,
+      },
+      host: {
+        id: this.hostId,
+        hostname: this.hostname,
+        ip: this.ip,
+        mac: this.macAddress,
+        os: this.os,
+      },
+    };
+  }
+
+  public generateAlert(ts: number, entityID?: string, parentEntityID?: string): AlertEvent {
```

Comment:
Does it make sense to give default values to `ts` and `entityID` in the function definition? `ts` could have a default value of `new Date().getTime()`